### PR TITLE
implementing missing pyproject on transforms

### PR DESCRIPTION
**Pull request by blublinsky**

### CODE REVIEW

File: transforms/universal/ededup/ray/Dockerfile [hunk 1]
```diff
@@ -10,8 +10,12 @@ RUN cd data-processing-lib-python && pip install --no-cache-dir -e .
 COPY --chown=ray:users data-processing-lib-ray/ data-processing-lib-ray/ 
 RUN cd data-processing-lib-ray    && pip install --no-cache-dir -e .
 
-COPY requirements.txt requirements.txt
-RUN pip install --no-cache-dir -r  requirements.txt
+# Install ray project source
+COPY --chown=ray:users src/ src/
+COPY --chown=ray:users pyproject.toml pyproject.toml
+COPY --chown=ray:users Readme.md Readme.md
+COPY --chown=ray:users images/ images/
```
Comment (daw3rd) [inline transforms/universal/ededup/ray/Dockerfile:17]: Not sure we need images.

File: transforms/universal/ededup/ray/Makefile [hunk 2]
```diff
@@ -22,38 +22,29 @@ test-src:: .transforms.test-src
...
+setup:: .transforms.setup
 
-run-cli-sample: 
-	$(MAKE) RUN_FILE=$(TRANSFORM_NAME)_transform_ray.py \
-                RUN_ARGS="--run_locally True --data_local_config \"{ 'input_folder' : '../test-data/input', 'output_folder' : '../output'}\"  \
-                --ededup_num_hashes 2"	\
-                .transforms.run-src-file
+# set the version of python transform that this depends on.
+set-versions:
+	$(MAKE) TRANSFORM_PYTHON_VERSION=${CODE2PARQUET_PYTHON_VERSION} TOML_VERSION=$(CODE2PARQUET_PYTHON_VERSION) .transforms.set-versions
```
Comment (daw3rd) [inline transforms/universal/ededup/ray/Makefile]: TOML_VERSION not required

File: transforms/universal/ededup/ray/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+[project]
+name = "dpk_ededup_transform_ray"
+version = "0.4.0.dev6"
+requires-python = ">=3.10"
+description = "code2parquet Ray Transform"
```
Comment (daw3rd) [inline transforms/universal/ededup/ray/pyproject.toml]: Ededup instead of code2parquet

File: transforms/universal/ededup/ray/Dockerfile (hunk 1 above)
Comment (blublinsky) [reply transforms/universal/ededup/ray/Dockerfile:17]: We do. Pyproject.toml references it

File: data-processing-lib/python/src/data_processing/test_support/abstract_test.py [hunk 4]
```diff
@@ -218,7 +218,7 @@ def __confirm_diffs(
                 # These are binary files.
                 da = DataAccessLocal()
                 f1_bytes = da.get_file(src)
-                f2_bytes = da.get_file(dest)
```
Comment (roytman) [inline data-processing-lib/python/src/data_processing/test_support/abstract_test.py:221]: should we compare the expected and destination?

File: transforms/universal/doc_id/ray/Dockerfile [hunk 5]
```diff
@@ -10,8 +10,11 @@ RUN cd data-processing-lib-python && pip install --no-cache-dir -e .
 COPY --chown=ray:users data-processing-lib-ray/ data-processing-lib-ray/ 
 RUN cd data-processing-lib-ray    && pip install --no-cache-dir -e .
 
-COPY requirements.txt requirements.txt
-RUN pip install --no-cache-dir -r  requirements.txt
+# Install ray project source
+COPY --chown=ray:users src/ src/
```
Comment (roytman) [inline transforms/universal/doc_id/ray/Dockerfile:14]: can it be done in the Makefile by `.defaults.copy-lib` ?

**Review by daw3rd — DISMISSED** [review]

File: transforms/universal/doc_id/ray/pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,45 @@
+[project]
+name = "dpk_docid_transform_ray"
+version = "0.4.0.dev6"
+requires-python = ">=3.10"
```
Comment (roytman) [inline transforms/universal/doc_id/ray/pyproject.toml:4]: should we use ">=3.10,<3.12" ?

File: transforms/universal/ededup/ray/pyproject.toml (hunk 3 above)
Comment (blublinsky) [reply transforms/universal/ededup/ray/pyproject.toml]: fixed

**Review by roytman — COMMENTED** [review]
some of the comments are relevant to other transformers

File: data-processing-lib/python/src/data_processing/test_support/abstract_test.py (hunk 4 above)
Comment (blublinsky) [reply data-processing-lib/python/src/data_processing/test_support/abstract_test.py:221]: nop, source and expected

File: transforms/universal/doc_id/ray/Dockerfile (hunk 5 above)
Comment (blublinsky) [reply transforms/universal/doc_id/ray/Dockerfile:14]: Its done there, but we still need to install

File: transforms/universal/doc_id/ray/pyproject.toml (hunk 6 above)
Comment (blublinsky) [reply transforms/universal/doc_id/ray/pyproject.toml:4]: Technically yes, but we are doing this way everywhere else

File: transforms/universal/doc_id/ray/Dockerfile (hunk 5 above)
Comment (roytman) [reply transforms/universal/doc_id/ray/Dockerfile:14]: yes we need to install it, but we don't have to copy each part separately, the temp directory contains only relevant files

File: transforms/universal/doc_id/ray/pyproject.toml (hunk 6 above)
Comment (roytman) [reply transforms/universal/doc_id/ray/pyproject.toml:4]: except kfp libs :-)

File: transforms/universal/ededup/ray/Makefile (hunk 2 above)
Comment (roytman) [reply transforms/universal/ededup/ray/Makefile]: indeed, `TOML_VERSION=$(DOCKER_IMAGE_VERSION)`

File: transforms/universal/doc_id/ray/Dockerfile (hunk 5 above)
Comment (blublinsky) [reply transforms/universal/doc_id/ray/Dockerfile:14]: wait, we are installing local directory. We do not need any makefiles support here

File: transforms/universal/ededup/ray/Makefile (hunk 2 above)
Comment (daw3rd) [reply transforms/universal/ededup/ray/Makefile]: I don't believe TOML_VERSION is needed.  Can you try removing it?

File: transforms/universal/doc_id/ray/pyproject.toml (hunk 6 above)
Comment (daw3rd) [reply transforms/universal/doc_id/ray/pyproject.toml:4]: Is this a transitive dependency from data-prep-toolkit and if so, can it be removed?

File: transforms/universal/doc_id/ray/Dockerfile (hunk 5 above)
Comment (daw3rd) [reply transforms/universal/doc_id/ray/Dockerfile:14]: Makefile infrastructure is not part of the docker build environment.  yes, we could copy in the local Makefile, but it references things outside of this directory.

File: transforms/universal/ededup/ray/Makefile (hunk 2 above)
Comment (blublinsky) [reply transforms/universal/ededup/ray/Makefile]: Its done

**Review by roytman — APPROVED** [review]
LGTM

**Pull request merged**